lib/possible_bipartition.rb

```diff
@@ -1,4 +1,88 @@
-
+# BFS
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+  
+  # when can we stop? when we visited all the dogs
+  # what about friendly dogs? they're exempt
+  friendlies = dislikes.select { |enemyList| enemyList == [] }
+  needToVisitThisMany = dislikes.length - friendlies.length
+  
+  dislikes.length.times do |i|
+    # starting a new scenario with i as the starterNode
+    queue = [i]
+    visited = {}
+    # visited.key = index# for each dog, visited.value = T/F for easy flipping back and forth
+    
+    visited[i] = true
+    
+    until queue.empty?
+      currDog = queue.pop()
+      newColor = !visited[currDog]
+      
+      dislikes[currDog].each do |enemy|
+        if visited[enemy].nil?
+          visited[enemy] = newColor 
+          queue << enemy
```

Comment:
You're shoveling and popping things from the same end, so this is really a `stack` instead of a queue.